### PR TITLE
HomeScreen setup: Create Database and Dao. Implement DI.

### PR DESCRIPTION
**Pull request by DurtDiv3r**
**Work Done:**
Setup database, dao and initial DI for later Paging implementation.

**Reference material:**
None

### CODE REVIEW

File: app/src/main/java/com/islaharper/dawnofandroid/di/DatabaseModule.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import javax.inject.Singleton
+
+@InstallIn(SingletonComponent::class)
+@Module
+object DatabaseModule {
+
+    @Provides
+    @Singleton
+    fun provideDatabase(@ApplicationContext context: Context): AndroidFlavourDb {
+        return Room.databaseBuilder(
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/di/DatabaseModule.kt]: Why would we write a `AndroidFlavourDb.create` factory function and then not use it in the Dagger provides function?

File: app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.islaharper.dawnofandroid.domain.model.Flavour
+
+@Dao
+interface AndroidFlavourDao {
+
+    @Query("SELECT * FROM flavour_table ORDER BY id ASC")
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt]:
```kotlin
flavour_table
```

Can you please see if there is a good convention to use for db table names?

**Review by drawers — COMMENTED** [review]
Looks good so far! Just a few minor comments

File: app/src/main/java/com/islaharper/dawnofandroid/di/DatabaseModule.kt (hunk 1 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/di/DatabaseModule.kt]: Updated commit... This was an artifact from writing unit test for the remote mediator

File: app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt (hunk 2 above)
Comment (drawers) [reply app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt]: not sure if appending `_table` to the table name is useful:

https://www.sqlshack.com/learn-sql-naming-conventions/

how about `flavour` ?

File: app/src/main/java/com/islaharper/dawnofandroid/domain/model/Flavour.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@ import com.islaharper.dawnofandroid.util.Constants
 import kotlinx.serialization.Serializable
 
 @Serializable
-@Entity(tableName = Constants.FLAVOUR_DB_TABLE)
+@Entity(tableName = Constants.FLAVOURS_TABLE)
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/domain/model/Flavour.kt]: Can we inline this (if it's only used in one place) or find somewhere more appropriate to put it so its encapsulated?

**Review by drawers — APPROVED** [review]
approved but I trust you to try and address the comments before merging

File: app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt (hunk 2 above)
Comment (DurtDiv3r) [reply app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt]: Hi, I was wondering about that myself and there was a fair bit of conflicting information on use of singular or plural and whether to use _table or not as even Android Developer codelab used _table: https://developer.android.com/codelabs/android-room-with-a-view-kotlin#4. Happy to rename table to flavour

Comment (drawers) [reply app/src/main/java/com/islaharper/dawnofandroid/data/local/dao/AndroidFlavourDao.kt]: Thank you that link is useful. Happy to go with whichever convention you want to use

**Pull request merged**